citest/azure_testing/az_contract.py

```diff
@@ -0,0 +1,135 @@
+# Create the base agent for Azure CITest scenario
+
+"""Support for specifying citest.json_contract.Contract on Azure resources."""
+
+# Python modules
+import json
+import logging
+import traceback
+
+# CITest Modules
+from .. import json_contract as jc
+from ..json_predicate import JsonError
+from ..service_testing import cli_agent
+
+
+class AzObjectObserver(jc.ObjectObserver):
+  """ Observe Az resources"""
+
+  def __init__(self, az, args, filter=None):
+    """Construct the observer.
+
+    Attributes:
+        az = AzCloudAgent instance to use.
+        args: Commang-line arguments list to execute.
+        filter: If provided, then use this to filter observations.
+    """
+
+    super(AzObjectObserver, self).__init__(filter)
+    self.__az = az
+    self.__args = args
+
+  def __str__(self):
+    return 'AzObjectObserver({0})'.format(self.__args)
+
+  def collect_observation(self, context, observation, trace=True):
+    args = context.eval(self.__args)
+    az_response = self.__az.run(args, trace=trace)
+    if not az_response.ok():
+        observation.add_error(
+            cli_agent.CliAgentRunError(self.__az, az_response))
+        return []
+
+    decoder = json.JSONDecoder()
+    try:
+      doc = decoder.decode(az_response.output)
+      if not isinstance(doc, list):
+        doc = [doc]
+      observation.add_all_objects(doc)
+      #self.filter_all_objects_to_observation(context, doc, observation)
```

Comment:
I havent been using filters in practice, but if you were to use a filter then you'd need this. The idea of the filters was to cut down on observed objects processed (and reported) downstream.